### PR TITLE
implement better ipfs running check

### LINKED ISSUE #244: Add better check if ipfs gateway is running
The current check only looks to see if the gateway port is taken. If so, we assume that ipfs is running.

However if another process is using that port, an no-descrptive error is thrown

`Unexpected token < in JSON at position 0`

### PR DESCRIPTION
**Pull request by perissology**
fixes #244

### CODE REVIEW

File: src/helpers/ipfs-daemon.js [hunk 1]
```diff
@@ -76,4 +77,27 @@ const setIPFSCORS = (ipfsRpc) => {
...
 }
 
-module.exports = { startIPFSDaemon, isIPFSCORS, setIPFSCORS }
+const isIPFSRunning = async (ipfsRpc) => {
+  const portTaken = await isPortTaken(ipfsRpc.port)
+
+  if (portTaken) {
+    if (!ipfsNode) ipfsNode = ipfsAPI(ipfsRpc)
+
+    const getId = promisify(ipfsNode.id)
```
Comment (0x6431346e) [inline src/helpers/ipfs-daemon.js]: `ipfsAPI` already returns a promise if a callback is not provided

File: src/helpers/ipfs-daemon.js [hunk 2]
```diff
@@ -76,4 +77,27 @@ const setIPFSCORS = (ipfsRpc) => {
...
+
+    try {
+      // if port is taken, attempt to fetch the node id
+      // if this errors, we can assume the port is taken
+      // by a process other then the ipfs gateway
+      await getId()
+      return true
+    } catch (e) {
+      console.log('error', e)
```
Comment (0x6431346e) [inline src/helpers/ipfs-daemon.js]: I would suggest we get rid of this, as showing two errors to the users can be confusing:
```
✔ Start a local Ethereum network
 ❯ Check IPFS
   ⠙ Start IPFS
     Add local files
   Publish app to APM
   Create DAO
   Open DAO
error SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at streamToValue (/home/daniel/repos/aragon/aragon-cli/node_modules/ipfs-api/src/utils/stream-to-json-value.js:25:18)
    at concat (/home/daniel/repos/aragon/aragon-cli/node_modules/ipfs-api/src/utils/stream-to-value.js:12:22)
    at ConcatStream.<anonymous> (/home/daniel/repos/aragon/aragon-cli/node_modules/concat-stream/inde ✔ Start a local Ethereum network
 ❯ Check IPFS
   ✖ Start IPFS
     → Starting IPFS failed: Error: serveHTTPApi: manet.Listen(/ip4/127.0.0.1/tcp/5001) failed: list…
     Add local files
   Publish app to APM
   Create DAO
   Open DAO
 ✖ Starting IPFS failed: Error: serveHTTPApi: manet.Listen(/ip4/127.0.0.1/tcp/5001) failed: listen tcp4 127.0.0.1:5001: bind: address already in use
```

File: src/helpers/ipfs-daemon.js (hunk 1 above)
Comment (perissology) [reply src/helpers/ipfs-daemon.js]: thanks

File: src/helpers/ipfs-daemon.js (hunk 2 above)
Comment (perissology) [reply src/helpers/ipfs-daemon.js]: oops, didn't mean to leave that in

**Pull request merged**